### PR TITLE
dnn: add model and data for attention layer implementation

### PR DESCRIPTION
**Pull request by fengyuentau**
https://github.com/opencv/opencv/pull/24476

## Script to generate VIT_B_32

```python
import torch # version: 1.13.1
import torchvision # version: 0.14.1

dummy_input = torch.randn(1, 3, 224, 224, device="cpu")
input_names = [ "input" ]
output_names = [ "output" ]

opset = 11

model = torchvision.models.vit_b_32(weights=torchvision.models.ViT_B_32_Weights.DEFAULT)
torch.onnx.export(model, dummy_input, "vit_b_32.opset{}.onnx".format(opset), verbose=False, export_params=True,
                  opset_version=opset, input_names=input_names, output_names=output_names)

from onnxsim import simplify # version: 0.4.33
import onnx # 1.14.1

model_sim, check = simplify("vit_b_32.opset{}.onnx".format(opset))
assert check, "Simplified model could not be validated"
onnx.save(model_simplified, "vit_b_32.onnx")
```

### CODE REVIEW

**Comment by dkurt** [comment]
Can you please specify a source of test data? Or changes to Python generate script missed?

**Comment by fengyuentau** [comment]
Oh, yes, sure, I forgot that. Will do later.

**Comment by fengyuentau** [comment]
Code is added.

File: testdata/dnn/onnx/models/object_tracking_vittrack_2023sep.onnx
Comment (dkurt) [inline testdata/dnn/onnx/models/object_tracking_vittrack_2023sep.onnx:1]: There is no script to generate this model and it's large. Please download by `download_models.py`

**Comment by dkurt** [comment]
@fengyuentau, please squash all the commits into one to make total patch size smaller

**Comment by fengyuentau** [comment]
> @fengyuentau, please squash all the commits into one to make total patch size smaller

Done.

**Pull request merged**